src/sip/transp.c

```diff
@@ -1459,17 +1509,35 @@ int sip_transp_send(struct sip_connqent **qentp, struct sip *sip, void *sock,
 		if (err)
 			return err;
 
-		if (connh)
-			connh(&laddr, dst, mb, arg);
-
 		if (!sock) {
-			transp = transp_find(sip, tp, sa_af(&dsttmp), &dsttmp);
+			transp = (struct sip_transport	*)
+				transp_find(sip, tp, sa_af(&dsttmp), &dsttmp);
 			if (!transp)
 				return EPROTONOSUPPORT;
 
+			if (!transp->sock) {
+				err = udp_listen((struct udp_sock **)
+						 &transp->sock, &transp->laddr,
+				 	 	 udp_recv_handler, transp);
```

Comment:
Can we avoid to open the UDP listen socket?

Reply:
Like discussed: Please remove the changes 1518-1534. For UDP it makes no difference if the listen socket is opened at startup or at first REGISTER.

Reply:
Yes, with UDP a listening socket has to be opened regardless, so there's not really a point in not opening the socket when adding the transport. Done.